provision/ec2.go

```diff
@@ -165,39 +166,70 @@ func (p *EC2Provisioner) CreateInstance(opts EC2CreateInstanceOptions) (*cfg.Rem
 		return nil, err
 	}
 
+	// Check response validity
+	if runResp.Instances == nil || len(runResp.Instances) == 0 {
+		return nil, errors.New("Unable to start instances: " + runResp.String())
+	}
+
 	// Loop until intance is running
-	fmt.Fprint(p.out, "Checking status of requested instance...")
-	attempts := 0
-	var instanceStatus *ec2.DescribeInstancesOutput
+	fmt.Fprintln(p.out, "Checking status of requested instance...")
+	var (
+		attempts = 0
+		instance ec2.Instance
+	)
 	for {
+		// Wait briefly between checks
+		time.Sleep(3 * time.Second)
 		attempts++
+
 		// Request instance status
 		result, err := p.client.DescribeInstances(&ec2.DescribeInstancesInput{
 			InstanceIds: []*string{runResp.Instances[0].InstanceId},
 		})
 		if err != nil {
 			return nil, err
 		}
-		if len(result.Reservations) == 0 || len(result.Reservations[0].Instances) == 0 {
+
+		// Check if reservations are present
+		if result.Reservations == nil || len(result.Reservations) == 0 ||
+			len(result.Reservations[0].Instances) == 0 {
```

Comment:
Is there something similar to ruby's [`#dig`](https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig)? That would simplify this a lot. I understand this is a struct but thought it could be nice to have since you are refactoring.

Reply:
The idiomatic way to do this in Go is to provide `Get()` functions, which Amazon has been kind enough to not provide. I think there's a dereference helper function coming in #296, but I'd rather just do it in one go if we do refactor this into a less verbose check - I think I'll leave it as it is for the time being

Reply:
This paradigm is interesting to: https://github.com/storozhukBM/verifier - thoughts?